src/nomad_parser_fhiaims/schema_packages/schema_package.py

```diff
@@ -0,0 +1,82 @@
+from typing import (
+    TYPE_CHECKING,
+)
+
+if TYPE_CHECKING:
+    from nomad.datamodel.datamodel import (
+        EntryArchive,
+    )
+    from structlog.stdlib import (
+        BoundLogger,
+    )
+
+from nomad.config import config
+from nomad.metainfo import SchemaPackage
+from nomad_simulations.schema_packages import general, model_method, model_system, outputs, properties
+from nomad.parsing.file_parser.mapping_parser import MappingAnnotationModel
+
+configuration = config.get_plugin_entry_point(
+    'nomad_parser_fhiaims.schema_packages:schema_package_entry_point'
+)
+
+m_package = SchemaPackage()
+
+TEXT_ANNOTATION_KEY = 'text'
+
+class Program(general.Program):
+
+    general.Program.version.m_annotations[TEXT_ANNOTATION_KEY] = MappingAnnotationModel(mapper='.version')
+
+
+class XCFunctional(model_method.XCFunctional):
+
+    model_method.XCFunctional.libxc_name.m_annotations[TEXT_ANNOTATION_KEY] = MappingAnnotationModel(mapper='.name')
+
+class DFT(model_method.DFT):
+
+    model_method.DFT.xc_functionals.m_annotations[TEXT_ANNOTATION_KEY] = MappingAnnotationModel(mapper=('get_xc_functionals', ['.controlInOut_xc']))
+
+
+class AtomicCell(model_system.AtomicCell):
+
+    model_system.AtomicCell.lattice_vectors.m_annotations[TEXT_ANNOTATION_KEY] = MappingAnnotationModel(mapper='.lattice_vectors')
+
+    model_system.AtomicCell.positions.m_annotations[TEXT_ANNOTATION_KEY] = MappingAnnotationModel(mapper='.structure.positions', unit='angstrom')
```

Comment:
This leads me to the question: what if the units are defined at the `Quantity` level?

Reply:
This is independent of the quantity unit, This is the unit used by the code.

Reply:
Sure, but I was trying to make a different point:
in the old approach, we would set these units in `TextParser.Quantity` and then pass them to the schema.
Is this still a viable option? Would `MappingAnnotation` pick this up?

Reply:
ah TextParser.Quantity, you can define it in textparser yes and would be picked up and when you define it again in mapping parser it would produce an error which would not handled correctly, thanks for pointing out i will provide a fix